Lib/test/test_tarfile.py

```diff
@@ -1899,6 +1899,48 @@ def test_pax_extended_header(self):
         finally:
             tar.close()
 
+    def test_create_pax_header(self):
+        # The ustar header should contain values that can be
+        # represented reasonably, even if a better (e.g. higher
+        # precision) version is set in the pax header.
+        # Issue #45863
+
+        # values that should be kept
+        t = tarfile.TarInfo()
+        t.name = "foo"
+        t.mtime = 1000.1
+        t.size = 100
+        t.uid = 123
+        t.gid = 124
+        info = t.get_info()
+        header = t.create_pax_header(info, encoding="iso8859-1")
+        self.assertEqual(info['name'], "foo")
+        # mtime should be rounded to nearest second
+        self.assertEqual(info['mtime'], 1000)
```

Comment:
If you want to ensure that you get an int, please add ``assertIsInstance(info['mtime'], int)``.

``1000 == 1000.0`` is true in Python ;-)